src/referenced-fields.ts

```diff
@@ -361,6 +361,7 @@ const countTableSelects = (
 		case 'Alias':
 		case 'Any':
 		case 'Average':
+		case 'Cast':
```

Comment:
Added Cast as an unary node - this should count in the case of `CAST(SELECT 1 FROM ....) AS BOOLEAN`